common/djangoapps/student/views.py

```diff
@@ -423,17 +423,23 @@ def _cert_info(user, course_overview, cert_status, course_mode):  # pylint: disa
                 )
 
     if status in {'generating', 'ready', 'notpassing', 'restricted', 'auditing', 'unverified'}:
+        cert_grade_percent = -1
+        persisted_grade_percent = -1
         persisted_grade = CourseGradeFactory().read(user, course=course_overview)
         if persisted_grade is not None:
-            status_dict['grade'] = unicode(persisted_grade.percent)
-        elif 'grade' in cert_status:
-            status_dict['grade'] = cert_status['grade']
-        else:
+            persisted_grade_percent = persisted_grade.percent
+
+        if 'grade' in cert_status:
+            cert_grade_percent = int(cert_status['grade'])
```

Comment:
That's a good catch. I think it does. The grade is stored as a 5-character string on the model, which to me implies that it can be a decimal.